File: medplum-java/medplum-model/src/main/java/com/fhir/MedicationKnowledge_AdministrationGuidelines.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationKnowledge_AdministrationGuidelines.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationKnowledge_AdministrationGuidelines {
  Optional<List<MedicationKnowledge_Dosage>> dosage();

  Optional<List<MedicationKnowledge_PatientCharacteristics>> patientCharacteristics();

  Optional<Reference> indicationReference();

  Optional<CodeableConcept> indicationCodeableConcept();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  static ImmutableMedicationKnowledge_AdministrationGuidelines.Builder builder() {
    return ImmutableMedicationKnowledge_AdministrationGuidelines.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MedicationKnowledge_AdministrationGuidelines.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { MedicationKnowledge_Dosage } from "./MedicationKnowledge_Dosage";
import { MedicationKnowledge_PatientCharacteristics } from "./MedicationKnowledge_PatientCharacteristics";
import { Reference } from "./Reference";

export interface MedicationKnowledge_AdministrationGuidelines {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    dosage: MedicationKnowledge_Dosage[] | null | undefined;
    indicationCodeableConcept: CodeableConcept | null | undefined;
    indicationReference: Reference | null | undefined;
    patientCharacteristics:
        | MedicationKnowledge_PatientCharacteristics[]
        | null
        | undefined;
}
